lib/money/money_parser.rb

```diff
@@ -1,31 +1,41 @@
+# parse a amount from a string
 class MoneyParser
-  # parse a amount from a string
+  MARKS = "\.,·’"
+  EXTRA_MARKS = "\s˙'"
+
   def self.parse(input, currency = nil)
     new.parse(input, currency)
   end
 
   def parse(input, currency = nil)
-    Money.new(extract_money(input.to_s), currency)
+    Money.new(extract_money(input.to_s, currency), currency)
   end
 
   private
-  def extract_money(input)
-    return Money.zero if input.to_s.empty?
 
-    amount = input.scan(/\-?[\d\.\,]+/).first
+  def extract_money(input, currency_iso = nil)
+    return '0' if input.to_s.empty?
+
+    amount = input.scan(/(-?[\d#{MARKS}][\d#{MARKS}#{EXTRA_MARKS}]*)/).first
+    return '0' unless amount
+    amount = amount.first.tr(EXTRA_MARKS, '')
 
-    return Money.zero if amount.nil?
+    *other_marks, last_mark = amount.scan(/[#{MARKS}]/)
+    other_marks.uniq!
```

Comment:
Why? `other_marks` doesn't seem to be used in this method

Reply:
ah yes, i removed the code that was using this (but will add it back). The idea was that if the other marks are equal to the last mark, then it's for sure a thousand separator. This helps in cases where the currency is not specified. Example `1.000.000`